File: twitterEngine/src/spl/stomp/protocol/MessageFrame.java
package spl.stomp.protocol;

/**
 * @name Message Frame
 * @author  Lidan Hifi
 * @author  Ken Saggy
 * 
 * MESSAGE stomp frame
 */
class MessageFrame extends StompFrame {
    public static final String SERVER_MESSAGE = "-1";
    
    public MessageFrame(String destination, String subscription, String messageId, String body) {
        this(String.valueOf(System.currentTimeMillis() / 1000L), destination, subscription, messageId, body);
    }
    
    public MessageFrame(String timestamp, String destination, String subscription, String messageId, String body) {
        super(StompCommand.MESSAGE, body);
        addFrameHeader("destination", destination);
        addFrameHeader("subscription", subscription);
        addFrameHeader("timestamp", timestamp);
        addFrameHeader("message-id", messageId);
    }
}

File: twitterClient/twitterClient/src/stomp/frames/MessageFrame.cpp
//
//  MessageFrame.cpp
//  Object to represent the MESSAGE frame type
//
//  Authors:    Ken Saggy
//              Lidan Hifi
//
//  Copyright 2013 Lidan Hifi, Ken Saggy

#include "../../../include/stomp/frames/MessageFrame.h"

namespace stomp {
    MessageFrame::MessageFrame(): StompFrame(MESSAGE_FRAME_NAME) { }
    
    MessageFrame::MessageFrame(std::string destination, std::string subscription, std::string messageId, std::string body): StompFrame(MESSAGE_FRAME_NAME, body) {
        addHeaderPair("destination", destination);
        addHeaderPair("subscription", subscription);
        addHeaderPair("message-id", messageId);
    }
    
    void MessageFrame::accept(FrameVisitor *visitor)
    {
        visitor->visit(this);
    }
}
